From dvrabel@arcom.com Thu Jan 19 09:52:35 2006
Message-ID: <43CFD1DB.4010900@arcom.com>
Date: Thu, 19 Jan 2006 17:52:27 +0000
From: David Vrabel <dvrabel@arcom.com>
To: Greg KH <gregkh@suse.de>
Cc: David Vrabel <dvrabel@arcom.com>
Subject: driver core: platform_get_irq*(): return -ENXIO on error

platform_get_irq*() cannot return 0 on error as 0 is a valid IRQ on some
platforms, return -ENXIO instead.

Signed-off-by: David Vrabel <dvrabel@arcom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/platform.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/base/platform.c
+++ gregkh-2.6/drivers/base/platform.c
@@ -61,7 +61,7 @@ int platform_get_irq(struct platform_dev
 {
 	struct resource *r = platform_get_resource(dev, IORESOURCE_IRQ, num);
 
-	return r ? r->start : 0;
+	return r ? r->start : -ENXIO;
 }
 EXPORT_SYMBOL_GPL(platform_get_irq);
 
@@ -98,7 +98,7 @@ int platform_get_irq_byname(struct platf
 {
 	struct resource *r = platform_get_resource_byname(dev, IORESOURCE_IRQ, name);
 
-	return r ? r->start : 0;
+	return r ? r->start : -ENXIO;
 }
 EXPORT_SYMBOL_GPL(platform_get_irq_byname);
 
